src/main/scala/dsptools/counters/Counters.scala

```diff
@@ -164,11 +165,13 @@ abstract class Counter(countParams: CountParams) extends Module {
       //    else Mux(wrap,wrapTo,nextInSeq)
     }
     else {
-      if(wrap == Bool(true)) {
-        wrapTo
-      } else {
-        nextInSeq
+      val nextThing = Wire(Bool())
```

Comment:
A hardware mux is different from compile-time configuration, even if you can assume optimizers down the line will do the same thing [b/c Mux select is a Lit].